core/src/main/scala/codesearch/core/index/RustIndex.scala

```diff
@@ -28,25 +29,21 @@ class RustIndex(val ec: ExecutionContext) extends LanguageIndex[CratesTable] wit
     ".git"
   )
 
-  def csearch(searchQuery: String,
-              insensitive: Boolean,
-              precise: Boolean,
-              sources: Boolean,
-              page: Int = 0): Future[(Int, Seq[PackageResult])] = {
-    val answers = runCsearch(searchQuery, insensitive, precise, sources)
-    verNames().map { verSeq =>
-      val nameToVersion = Map(verSeq: _*)
-      (answers.length,
-       answers
-         .slice(math.max(page - 1, 0) * 100, page * 100)
-         .flatMap(uri => contentByURI(uri, nameToVersion))
-         .groupBy(x => (x._1, x._2))
-         .map {
-           case ((name, packageLink), results) =>
-             PackageResult(name, packageLink, results.map(_._3))
-         }
-         .toSeq
-         .sortBy(_.name))
+  override def csearch(args: SearchArguments, page: Int = 0): Future[CSearchPage] = {
```

Comment:
You're going to merge all the implementations of `csearch` in some other PR, right?

Reply:
Yes, Rust has a special method `contentByURI` - after complete #74 I can merge all contentByURI into the generic method - I remove this implementation `csearch`